src/aot/aot.cpp

```diff
@@ -165,6 +172,59 @@ std::optional<std::vector<uint8_t>> generate_section(
   return out;
 }
 
+// Clones the shim to final destination while also injecting
+// the custom .btaot section.
+int build_binary(const std_filesystem::path &shim,
```

Comment:
It feel like this could just return a bool to indicate success or failure as the only useful err is from `std::system(cmd)`, which could just be logged with `LOG(ERROR)` but not a big deal either way.

Reply:
Ah yeah that's simpler. Let's do `bool`